storage/replica.go

```diff
@@ -1072,6 +1076,51 @@ func (r *Replica) maybeGossipConfigsLocked(match func(configPrefix proto.Key) bo
 	}
 }
 
+// maybeGossipSystemMap scans the entire SystemDB span and gossips it.
+// The first call is on NewReplica. Further calls come from the trigger
+// on an EndTransactionRequest.
+//
+// Note that maybeGossipSystemMap gossips information only when the
+// lease is actually held. The method does not request a leader lease
+// here since LeaderLease and applyRaftCommand call the method and we
+// need to avoid deadlocking in redirectOnOrObtainLeaderLease.
+// TODO(tschottdorf): Can possibly simplify.
+func (r *Replica) maybeGossipSystemMap() {
+	r.Lock()
+	defer r.Unlock()
+	r.maybeGossipSystemMapLocked()
+}
+
+func (r *Replica) maybeGossipSystemMapLocked() {
+	if r.rm.Gossip() == nil || !r.isInitialized() {
+		return
+	}
+
+	if lease := r.getLease(); !lease.OwnedBy(r.rm.RaftNodeID()) || !lease.Covers(r.rm.Clock().Now()) {
+		// Do not gossip when a leader lease is not held.
+		return
```

Comment:
if the leader dies after committing the batch containing `EndTransaction` but before the trigger is executed (heck, or even dies before any of the new gossip'ed info makes it out of the node) or the lease just expired it can happen that you never gossip this new info. There should be a loop gossiping this at intervals somewhere (`Store` already has/had a few of those - `maybeGossipConfigs` etc).